### DISCUSSION TITLE
What is the easiest recommend way to deal with currencies for q-inputs

### DISCUSSION THREAD

**Discussion by b0ot** [quasarframework/quasar#9297] [General - Components / Directives / etc]
Objective: I'm looking for an easy way to deal with various currencies / decimals and formatting within Quasar 1.X. Essentially I want the user to be able to input a value of money and have it displayed nicely as they do so.

At first I tried:
[v-money](https://www.npmjs.com/package/v-money) However there were a number of posts on the forum recommending against it in favor of [vue-autonumeric](https://github.com/autoNumeric/vue-autoNumeric)

However installation / using vue-autoNumeric was fairly complicated to me, and after a few hours of trying to get it to work I've run out of ideas to try.

I also was considering [Q-Decimal](https://github.com/TobyMosque/app-extension-qdecimal) however this seemed fairly new.

Does anyone have any modern recommended approaches?
If it is vue-autoNumeric, can you walk through the install process:

**Comment by b0ot** [comment]
Q-Decimal definitely was the easiest. It is a big glitchy but works well enough and shows some real potential.

**Comment by AdagioB** [comment]
I also started out with v-money, but quickly realized that it wasn't useful at all, if you want to be able to support both 0 and null. 
I went with vue-autonumeric. It was a bit tricky to get started, but ended up with something that works fine. My biggest issue with it, is that I can't set focus to it
What is your problems with vue-autonumeric? I'm using Quasar with UMD, where it was fairly easy to get started. I have no experience with the other types

**Comment by CynthiaBlue** [comment]
I'm also looking for a nice, easy way to deal with currency in Quasar.  It looks like v-money is no longer supported.

It really seems like there should be an easy, simple money mask built into quasar's input field.  Maybe that will happen in the future?

I'll look into vue-autonumeric and Q-Decimal

**Reply by pdanpdan** [reply]
This one looks pretty nice
https://dm4t2.github.io/vue-currency-input/playground.html

**Reply by AdagioB** [reply]
@pdanpdan 
Looks great and according to documentation it should also work in Vue 3. Unfortunately it doesn't look like I'll be able to use it as in my project I can't use a package installer. Is there a way to link directly to a js file? I can't seem to find anything

**Reply by CynthiaBlue** [reply]
> This one looks pretty nice https://dm4t2.github.io/vue-currency-input/playground.html

These examples assume a person really knows a lot more than I know. I'm trying to piecemeal this into my code.

LIke the playground, it gives me:
```
{
  "currency": "USD",
  "currencyDisplay": "symbol",
  "hideCurrencySymbolOnFocus": true,
  "hideGroupingSeparatorOnFocus": true,
  "hideNegligibleDecimalDigitsOnFocus": true,
  "autoDecimalDigits": false,
  "autoSign": true,
  "useGrouping": true,
  "accountingSign": false
}
```

But where in the world does that go?

**Reply by MilosPaunovic** [reply]
The first thing then is to study this:
[https://dm4t2.github.io/vue-currency-input/guide.html](https://dm4t2.github.io/vue-currency-input/guide.html)

**Reply by CynthiaBlue** [reply]
> The first thing then is to study this: https://dm4t2.github.io/vue-currency-input/guide.html

I've been reading that... when i implement the examples, I get this error:
Cannot assign to read only property '0' of string 'currency'

And I cannot figure out why.

**Reply by MilosPaunovic** [reply]
They have `Quasar` example on their docs page:
[https://dm4t2.github.io/vue-currency-input/examples.html#usage-with-quasar-veevalidate](https://dm4t2.github.io/vue-currency-input/examples.html#usage-with-quasar-veevalidate)

Start from there.

**Comment by pdanpdan** [comment]
You mean this?
https://vue-currency-input-v1.netlify.app/guide/#direct-download-via-cdn

**Reply by AdagioB** [reply]
Ah yes, blind stupid me. Thanks, just what I needed
